### PR TITLE
[Usability] mxnet version check

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##

@kaonashi-tyc 

MXNet with unsupported versions will trigger exceptions. 

```
>>> import gluonnlp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/haibilin/gluon-nlp/src/gluonnlp/__init__.py", line 49, in <module>
    utils.version.check_version('1.6.0', library=mxnet)
  File "/Users/haibilin/gluon-nlp/src/gluonnlp/utils/version.py", line 49, in check_version
    raise AssertionError(msg)
AssertionError: Installed mxnet version 1.5.1 does not satisfy the minimum required version 1.6.0
```

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by eric-haibin-lin** [comment]
https://github.com/dmlc/gluon-nlp/issues/1059

**Comment by eric-haibin-lin** [comment]
@leezu I'm also switching from `cpu/py3` to `cpu/py3-master` in `ci/jenkins/Jenkinsfile_py3-master_gpu_doc`. Let me know if there's any concern

**Comment by leezu** [comment]
@eric-haibin-lin why change? The actual environment inside the AWS Batch job depends on the env file in `docker`. Was there any issue with the existing environment?

**Comment by eric-haibin-lin** [comment]
The first commit fails the doctest: https://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-master-gpu-doc/detail/PR-1063/1/pipeline

**Comment by eric-haibin-lin** [comment]
I am not sure why it complains about `notedown` not found. https://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-master-gpu-doc/detail/PR-1063/8/pipeline @leezu any idea?

File: ci/jenkins/Jenkinsfile_py3-master_gpu_doc [hunk 1]
```diff
@@ -67,10 +67,10 @@ core_logic: {
               if (env.BRANCH_NAME.startsWith('PR-')){
                 sh """
                 set +e
-                conda activate ./conda/cpu/py3
+                conda activate ./conda/cpu/py3-master
 
                 python3 ci/batch/submit-job.py --region us-east-1 --wait \
-                  --timeout 1800 --saved-output ./docs/examples --conda-env docker/py3 \
+                  --timeout 1800 --saved-output ./docs/examples --conda-env docker/py3-master \
```
Comment (leezu) [inline ci/jenkins/Jenkinsfile_py3-master_gpu_doc]: `docker/py3-master` does not exist. You need to update the `docker/py3` file instead.
We only run one job for AWS Batch. Not one for master and stable.

File: env/cpu/py3-master.yml [hunk 2]
```diff
@@ -26,3 +26,21 @@ dependencies:
     - spacy==2.2.2
     - sentencepiece==0.1.83
     - sphinx-autodoc-typehints==1.7.0
+    - notedown
+    - recommonmark
+    - sphinx_gallery
+    - sphinx_rtd_theme
+    - recommonmark
+    - pandoc=1.19.2
```
Comment (leezu) [inline env/cpu/py3-master.yml]: Should be `==`. Also for the version specifiers below.

File: env/cpu/py3-master.yml [hunk 3]
```diff
@@ -26,3 +26,21 @@ dependencies:
     - spacy==2.2.2
     - sentencepiece==0.1.83
     - sphinx-autodoc-typehints==1.7.0
+    - notedown
```
Comment (leezu) [inline env/cpu/py3-master.yml]: Can these packages be removed from the `cpu/py3` build then?

File: .pytype.cfg [hunk 4]
```diff
@@ -5,4 +5,4 @@ inputs =
     src/gluonnlp
 
 # Python version (major.minor) of the target code.
-python_version = 3.5
+python_version = 3.6
```
Comment (leezu) [inline .pytype.cfg]: Why?

**Comment by eric-haibin-lin** [comment]
@leezu I came across the error below in https://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-master-cpu-unittest/detail/PR-1063/29/pipeline for commit 25d1449:
```
[2019-12-25T09:58:57.547Z] UnsatisfiableError: The following specifications were found to be in conflict:

[2019-12-25T09:58:57.547Z]   - pytest=4.5.0 -> python[version='>=3.7,<3.8.0a0'] -> openssl[version='>=1.1.1a,<1.1.2a']

[2019-12-25T09:58:57.547Z]   - python=3.5

```
But I found that the dependency is the same in `cpu/py3`...

File: env/cpu/py3-master.yml [hunk 5]
```diff
@@ -4,26 +4,39 @@ dependencies:
   - python=3.5
   - pip
   - perl
+  - pylint
```
Comment (leezu) [inline env/cpu/py3-master.yml]: We can't use conda to install these packages for python 3.5 due to bugs in conda. Let's use pip instead.

But for simplicity, let's merge the Python 3.5 CI fixes as part of https://github.com/dmlc/gluon-nlp/pull/1023 Then you don't need to bother about them in this PR

Comment (eric-haibin-lin) [reply env/cpu/py3-master.yml]: It looks like the master branch CI is failing with lots of numba errors

Comment (leezu) [reply env/cpu/py3-master.yml]: That's due to the bug's in conda. Conda is not installing the most recent version of numba for some reason if you specify Python 3.5.

Just remove all changes to the CI environment files from your PR and it should pass.

**Comment by mli** [comment]
Job PR-1063/40 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1063/40/index.html
